src/main/ruby/core/truffle/kernel_operations.rb

```diff
@@ -22,82 +22,83 @@ def self.define_read_only_global(name, getter)
+      :$/,
+      -> { global_variable_get(:$/) },
+      -> v { if v && !Truffle::Type.object_kind_of?(v, String)
+               raise TypeError, '$/ must be a String'
+             end
+             global_variable_set(:$/, v) })
 
-$, = nil # It should be defined by the time boot has finished.
+    $/ = "\n".freeze
 
-$= = false
+    alias $-0 $/
 
-Truffle::KernelOperations.define_hooked_variable(
-  :$VERBOSE,
-  -> { Truffle::KernelOperations.global_variable_get(:$VERBOSE) },
-  -> v { v = if v.nil?
-               nil
-             else
-               !!v
+    define_hooked_variable(
+      :$,,
+      -> { global_variable_get(:$,) },
+      -> v { if v && !Truffle::Type.object_kind_of?(v, String)
+               raise TypeError, '$, must be a String'
              end
-         Truffle::KernelOperations.global_variable_set(:$VERBOSE, v) })
-
-$VERBOSE = case Truffle::Boot.get_option 'verbosity'
-           when :TRUE
-             true
-           when :FALSE
-             false
-           when :NIL
-             nil
-           end
-
-alias $-v $VERBOSE
-alias $-w $VERBOSE
-
-Truffle::KernelOperations.define_hooked_variable(
-  :$stdout,
-  -> { Truffle::KernelOperations.global_variable_get(:$stdout) },
-  -> v { raise TypeError, "$stdout must have a write method #{v.class} given." unless v.respond_to?(:write)
-         Truffle::KernelOperations.global_variable_set(:$stdout, v) })
-
-alias $> $stdout
-
-Truffle::KernelOperations.define_hooked_variable(
-  :$stderr,
-  -> { Truffle::KernelOperations.global_variable_get(:$stderr) },
-  -> v { raise TypeError, "$stderr must have a write method #{v.class} given." unless v.respond_to?(:write)
-         Truffle::KernelOperations.global_variable_set(:$stderr, v) })
+             global_variable_set(:$,, v) })
+
+    $, = nil # It should be defined by the time boot has finished.
+
+    $= = false
+
+    define_hooked_variable(
+      :$VERBOSE,
+      -> { global_variable_get(:$VERBOSE) },
+      -> v { v = if v.nil?
+                   nil
+                 else
+                   !!v
+                 end
+             global_variable_set(:$VERBOSE, v) })
+
+    $VERBOSE = case Truffle::Boot.get_option 'verbosity'
+               when :TRUE
+                 true
+               when :FALSE
+                 false
+               when :NIL
+                 nil
+               end
+
+    alias $-v $VERBOSE
+    alias $-w $VERBOSE
+
+    define_hooked_variable(
+      :$stdout,
+      -> { global_variable_get(:$stdout) },
+      -> v { raise TypeError, "$stdout must have a write method #{v.class} given." unless v.respond_to?(:write)
+             global_variable_set(:$stdout, v) })
+
+    alias $> $stdout
+
+    define_hooked_variable(
+      :$stderr,
+      -> { global_variable_get(:$stderr) },
+      -> v { raise TypeError, "$stderr must have a write method #{v.class} given." unless v.respond_to?(:write)
+             global_variable_set(:$stderr, v) })
+  end
```

Comment:
You reordered them but it's same set of globals defined here, right? (The diff makes it hard to see)

Reply:
Yes, it's the same set of globals.